server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -219,13 +240,24 @@ public void run() {
         // towards their goal
         iter = store.iterator();
         while (iter.hasNext()) {
-          TabletLocationState tls = iter.next();
-          if (tls == null) {
+          final ManagerTabletInfo mti = iter.next();
+          if (mti == null) {
+            LOG.debug("ManagerTabletInfoIterator returned null ManagerTabletInfo");
```

Comment:
Is it returning null expected?

Reply:
I modified this to throw an IllegalStateException in 9153726